components/Common/CodeBox/index.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+  DocumentDuplicateIcon,
+  CodeBracketIcon,
+} from '@heroicons/react/24/outline';
+import classNames from 'classnames';
+import type { FC, PropsWithChildren } from 'react';
+import { useRef } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import Button from '@/components/Common/Button';
+import Notification from '@/components/Common/Notification';
+import { useCopyToClipboard } from '@/hooks/useCopyToClipboard';
+
+import styles from './index.module.css';
+
+const languageMap: Record<string, string[]> = {
+  JSX: ['jsx'],
+  JSON: ['json'],
+  XML: ['xml'],
+  YML: ['yml', 'yaml'],
+  Bash: ['bash', 'sh', 'shell', 'zsh', 'shellscript'],
+  Shell: ['console', 'shellsession'],
+  TypeScript: ['ts', 'typescript'],
+  JavaScript: ['cjs', 'mjs', 'js', 'javascript'],
+};
+
+const labelsMap = Object.fromEntries(
```

Comment:
I don't like mappings being done (function calls) at root level of a file, it's not practical and pollutes global context. Transform this into a function if needed.